x-pack/plugins/notifications/server/services/types.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export interface EmailService {
+  sendPlainTextEmail(payload: PlainTextEmail): Promise<void>;
+}
+
+interface RelatedSavedObject {
+  id: string;
+  type: string;
+  namespace: string;
+}
+
+export type SenderContext = RelatedSavedObject[];
+
+export interface PlainTextEmail {
+  to: string[];
+  subject: string;
+  message: string;
+  context?: SenderContext;
```

Comment:
To be more future-proof, should we change the context to have the related objects be a property of it instead? It would allow to enhance the context with additional infos later if necessary?

```ts
context: RelatedSavedObject[];
```

to 

```ts
context: {
 relatedObjects: RelatedSavedObject[];
}
```

Reply:
That's a good one, I'll update it!